### ISSUE TITLE
Finding `.gitattributes` takes a very long time in large repos

### ISSUE THREAD

**Issue by joshfriend** [git-lfs/git-lfs#5897]
**Describe the issue**
We use git lfs in a repo with ~120k files tracked. Switching branches takes about 8 seconds and is mostly spent in `git lfs post-checkout`. Using `GIT_TRACE=true` the slowdown is easily isolatable to an `ls-files` operation that is run to locate `.gitattributes` files.

**System environment**
MacOS

**Output of `git lfs env`**
```
Endpoint=https://github.com/squareup/redacted.git/info/lfs (auth=none)
  SSH=redacted@github.com:squareup/redacted.git
LocalWorkingDir=/Users/jfriend/redacted
LocalGitDir=/Users/jfriend/redacted/.git
LocalGitStorageDir=/Users/jfriend/redacted/.git
LocalMediaDir=/Users/jfriend/redacted/.git/lfs/objects
LocalReferenceDirs=
TempDir=/Users/jfriend/redacted/.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneVerifyUnreachableAlways=false
PruneRemoteName=origin
LfsStorageDir=/Users/jfriend/redacted/.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=/opt/homebrew/opt/git/libexec/git-core
GIT_PS1_SHOWDIRTYSTATE=true
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Additional context**
 Ours `.gitattributes` file is stored in the root directory of the repo and we do not include any other `.gitattributes` files in subfolders of the repo. It would be a huge improvement for us if we could configure lfs to check _only the root .gitattributes_ file.

**Comment by chrisd8088** [comment]
Hey, thanks for the report!

I believe this is the same issue discussed in #5510, and in particular in https://github.com/git-lfs/git-lfs/issues/5510#issuecomment-2403020311 and my (lengthy) analysis in https://github.com/git-lfs/git-lfs/issues/5510#issuecomment-2424713847, so I'm going to close this issue unless there's a particular reason why it's different from the existing one.

---

One note which might possibly be of assistance, from the comment I linked above:

> if your workflow permits you to accept Git LFS leaving unlocked lockable files as writable files, instead of trying to make them read-only, the most direct way to avoid this performance issue would be to set the Git configuration [option](https://github.com/git-lfs/git-lfs/blob/3fd88b17a4549480211dc563e18c71002490b376/docs/man/git-lfs-config.adoc?plain=1#L392-L399) `lfs.setLockableReadOnly` to `false` (or set the `GIT_LFS_SET_LOCKABLE_READONLY` environment variable to `0` or `false`, etc.)

The search for `.gitattributes` files in the Git index isn't especially slow, I believe, but by default Git LFS also needs to look for `.gitattributes` files in the working tree, and that's why the `--others` option is used in the call to `git ls-files`.  Performing that search of the file hierarchy (as compared to searching Git's optimized internal index) can be slow with large repositories.

If you disable the `lfs.setLockableReadOnly` option, Git LFS won't try to unset the write permissions on lockable files, and can therefore skip this search for `.gitattributes` files entirely in the post-checkout and other hooks.

---

> It would be a huge improvement for us if we could configure lfs to check _only the root .gitattributes_ file.

The challenge here is that we'd then deviate from how Git operates, as it doesn't offer such a configuration option (to my knowledge).  We try hard to avoid these kinds of variations, as they can introduce unexpected behaviour and traps for the unwary.

Still, if someone wants to put together a PR for such a configuration option, along with tests and documentation, and works through all the edge cases which might come up for users, we'd certainly take a look at such a proposal.

I would suggest, though, that the kinds of concerns I outlined in https://github.com/git-lfs/git-lfs/issues/5510#issuecomment-2424713847 regarding the issues that arise if Git LFS treats `.gitattributes` files differently than Git would be the sorts of things we'd want to consider very carefully in the case of an option to ignore some but not all `.gitattributes` files.
